aiohttp/web_request.py

```diff
@@ -510,19 +510,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads, wrong_format=None, if_check=True):
-        """
-        Return BODY as JSON.
-        If the `Content-Type` is not `application/json`,
+    def json(self, *, loads=json.loads, allowed_types=None, err_handler=None):
+        """ Return BODY as JSON. If the `Content-Type` is not in allowed_types,
         then the request is rejected.
-        You can also handle it manually by passing the
-        `wrong_format` parameter.
+        You can also handle it manually by passing the `err_handler` parameter.
         """
-        if if_check and self.content_type != 'application/json':
-            if wrong_format is None:
+        wrong_format_flag = False
+        if isinstance(allowed_types, str):
```

Comment:
I don't like global switches. You have argument and it's default. `None` disables the check. For awhile it will be default to preserve BC. After some release it becomes `application/json` in defaults and may be we could prohibit `None` usage, but that's another story.